### PR TITLE
 feat: toBeDisabled custom matcher (#18)

### PR DESCRIPTION
**Pull request by callada**
**What**: toBeDisabled custom matcher

**Why**: Details #18

**How**: By testing whether the element is one of form elements and is disabled itself or has a disabled ancestor.

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Ready to be merged 
* [x] Added myself to contributors table

### CODE REVIEW

File: src/__tests__/index.js [hunk 1]
```diff
@@ -205,3 +205,101 @@ test('.toBeVisible', () => {
...
+
+      <fieldset disabled={true}>
+        <button data-testid="inherited-element">x</button>
+      </fieldset>
+      <div disabled={true}>
+        <button data-testid="div-child-element">x</button>
+      </div>
+
+      <fieldset disabled={true} data-testid="fieldset-element">
+        <div><div><span>
```
Comment (kentcdodds) [inline src/__tests__/index.js]: Can you explain the purpose/value in these extra divs and spans?

Comment (callada) [reply src/__tests__/index.js]: To make sure, that it's "deep enough". I'll correct this, it's a bit silly as I think of it right now. 
Good catch, thank you :thumbsup:

**Comment by kentcdodds** [comment]
I think I'll happily merge this. I think @gnapse is busy. Looks like there are merge conflicts though. To fix the all-contributors one, you can simply manually fix the `.all-contributorsrc` and then run `git commit --amend --no-edit` and it should automatically regenerate the `README.md` (or you could run `npx kcd-scripts contributors generate`).

Thanks!

**Comment by gnapse** [comment]
I was on vacations without internet access for two weeks. I'm happy to see there's been a lot of activity in this repo and sad that I missed part of the action. I'll catch up and merge this and take care of all other issues in my notifications in the next few days. Thanks for your contribution!

**Comment by kentcdodds** [comment]
Welcome back @gnapse! I hope the vacation was super! Yeah! A lot has happened while you were gone!

**Comment by gnapse** [comment]
@callada one more thing: can you please take care of the merge conflicts? Let me know if you need help with those anyway. Thanks!

**Comment by callada** [comment]
@gnapse The same, I'm also no native English speaker. Thank you for review, I'll correct this and merge tomorrow.

File: README.md [hunk 2]
```diff
@@ -228,6 +228,33 @@ expect(container.querySelector('strong')).not.toBeVisible()
 // ...
 ```
 
+### `toBeDisabled`
+
+This allows you to check whether an element is disabled from user's perspective.
+It matches if the element is a form element and has or inherits `disabled`
+attribute.
```
[the remark this answers is not in the document]
Comment (antoaravinth) [reply README.md]: Also, in the issue link: https://github.com/gnapse/jest-dom/issues/18, I have seen only these elements can be disabled(https://html.spec.whatwg.org/multipage/semantics-other.html#disabled-elements). Can we add them in the README as well? That's gonna help our users.

[the remark this answers is not in the document]
Comment (gnapse) [reply README.md]: I think we can add that very link 👍 

@callada what do you think?

[the remark this answers is not in the document]
Comment (gnapse) [reply README.md]: Actually I also think we should have in the code the link to the spec about that complex rule about when a fieldset is considered disabled: https://html.spec.whatwg.org/multipage/form-elements.html#concept-fieldset-disabled

**Comment by callada** [comment]
Updated docs, added links. Plase, take a look if everything is ok.

**Comment by smacpherson64** [comment]
Looks really good!

**Pull request merged**